### ISSUE TITLE
client.release and pitfall with client.end are not documented in pg.Client

### ISSUE THREAD

**Issue by js-choi** [brianc/node-postgres#2101]
The [API documentation for `pg.Client`][] does not document `client.release`. The method is documented elsewhere in the [overview of pooling][] and the [API documentation for `pg.Pool`][] (see https://github.com/brianc/node-postgres/issues/1302#issuecomment-562372867), but, due to my inattentive blindness, I had missed that the examples used `client.release`, not `client.end`. It would be good if `client.release` itself was documented as a method on the [API documentation for `pg.Client`][], with its own section—even if it simply referred to the section about `releaseCallback` on the [API documentation for `pg.Pool`][].

I ran into this when I had assumed that I could call `client.end` to release a pool’s connection back to the pool, before I even had realized that `client.release` existed. Calling `pool.end` after calling `client.end` will hang the process (see https://github.com/brianc/node-postgres/issues/1414#issuecomment-322624860). It would be good if this pitfall was also documented for `pg.Client` in `client.end`’s section:
```js
const pg = require('pg');

async function f (callback) {
  const pool = new pg.Pool;
  const connection = await pool.connect();
  await callback(connection);
  await connection.end(); // This should be `await connection.release();`.
  await pool.end(); // Commenting this line would also prevent the hang.
}

(async () => {
  await f(async connection => {
    console.log('X');
  });
  console.log('Y'); // Never runs unless line 8 is commented.
})();
```
[API documentation for `pg.Client`]: https://node-postgres.com/api/client
[API documentation for `pg.Pool`]: https://node-postgres.com/api/pool
[overview of pooling]: https://node-postgres.com/features/pooling
[Calling `pool.end` after calling `client.end` will hang the process]: https://gist.github.com/js-choi/2024064faeeaafb79e24c6cdec347e22

**Comment by brianc** [comment]
I agree this is confusing.  I waffled back and forth internally on monkey-patching over `client.end` on a pooled client to return it to the pool, versus closing it, since you almost never want to close a pooled client.  I'll work on making the docs more explicit here.  It might be useful to add a `client.release` noop function on the base client as well just to avoid it exploding if you call that on a non-pooled client.  I also think calling `client.end` on a pooled client should throw a warning or something in your console....but I'll need to think about that bit a bit more.  Anyways...I'll work on making the docs better...thanks for your feedback!
